coreMacros/src/main/scala-3/me/shadaj/scalapy/py/Facades.scala

```diff
@@ -0,0 +1,174 @@
+    val classDynamicSymbol = Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
+    val classReaderSymbol = Symbol.requiredClass("me.shadaj.scalapy.readwrite.Reader")
+    val classWriterSymbol = Symbol.requiredClass("me.shadaj.scalapy.readwrite.Writer")
+    val classAnySymbol = Symbol.requiredClass("me.shadaj.scalapy.py.Any")
+    val methodFromSymbol = Symbol.requiredMethod("me.shadaj.scalapy.py.Any.from")
+
+    val readerTypeRepr = TypeIdent(classReaderSymbol).tpe
+    val writerTypeRepr = TypeIdent(classWriterSymbol).tpe
+    val dynamicTypeRepr = TypeIdent(classDynamicSymbol).tpe
+
+    val evidenceForDynamic = searchImplicit(readerTypeRepr.appliedTo(dynamicTypeRepr))
+    val evidenceForTypeT = searchImplicit(readerTypeRepr.appliedTo(TypeTree.of[T].tpe))
+
+    val methodSymbol = Symbol.spliceOwner.owner
+    val methodName = methodSymbol.name
+    val refss = methodSymbolParameterRefs(methodSymbol)
+
+    if refss.length > 1 then
+      report.throwError(s"method $methodName has curried parameter lists.")
+
+    val args = refss.headOption.toList.flatten
+
+    if (args.isEmpty) {
+      val selectDynamicAST = Apply(TypeApply(Select.unique(Apply(Select.unique(Apply(TypeApply(
+        Select.unique(resolveThis,"as"),List(TypeIdent(classDynamicSymbol))),List(evidenceForDynamic)),  
+        "selectDynamic"),List(Expr(methodName).asTerm)),"as"), List(TypeTree.of[T])),List(evidenceForTypeT)) 
+      selectDynamicAST.asExprOf[T]
+    }
+    else {
+      def constructASTforMethodFrom(arg: quotes.reflect.Term) = {
+        val argumentType = arg.tpe.typeSymbol
+        val applyArgTypeToWriter = writerTypeRepr.appliedTo(TypeIdent(argumentType).tpe)
+        val tree = Apply(Apply(TypeApply(Ref(methodFromSymbol),List(TypeIdent(argumentType))),List(arg)),
+          List(searchImplicit(applyArgTypeToWriter)))
+        tree
+      }
+
+      val typedVarargs = Typed(Inlined(None, Nil, Repeated(args.map(arg => constructASTforMethodFrom(arg)),
+        TypeIdent(classAnySymbol))),Applied(TypeIdent(defn.RepeatedParamClass),List(TypeIdent(classAnySymbol))))
+
+      val applyDynamicAST = Apply(TypeApply(Select.unique(Apply(Apply(Select.unique(Apply(TypeApply(
+        Select.unique(resolveThis,"as"),List(TypeIdent(classDynamicSymbol))),List(evidenceForDynamic)),
+        "applyDynamic"),List(Expr(methodName).asTerm)),List(typedVarargs)),"as"),List(TypeTree.of[T])),
+        List(evidenceForTypeT))
+
+      applyDynamicAST.asExprOf[T]
+    }
+  }
+
+  def native_named_impl[T: Type](using Quotes): Expr[T] = {
+    import quotes.reflect.*
+
+    def searchImplicit(typeReprParameter: TypeRepr): Term = {
+      Implicits.search(typeReprParameter) match {
+        case success: ImplicitSearchSuccess => {
+          success.tree
+        }
+        case _ => {
+          report.throwError(s"There is no implicit for ${typeReprParameter.show}")
+        }
+      }
+    }
+
+    val classDynamicSymbol = Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
```

Comment:
I tried to remove duplication and the only way that I found out at the moment is to rewrite in this way:
```
object SharedValues {
  def classDynamicSymbol(using Quotes) = {
    import quotes.reflect.*
    Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
  } 

  def classReaderSymbol(using Quotes) = {
    import quotes.reflect.*
    Symbol.requiredClass("me.shadaj.scalapy.readwrite.Reader")
  }
 }
 ```
 But we still have duplication of `import quotes.reflect.*` and I am not sure if it is okay
 
 I also tried to write like this: 
 ```
 class Share(using Quotes) {
  import quotes.reflect.*
  val classDynamicSymbol = quotes.reflect.Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
}
 ```
 But when I tried to compile it I received this error:
 ```
non-private value classDynamicSymbol in class Share refers to private given instance x$1
[error]    |in its type signature Share.this.x$1.reflect.Symbol
 ```

Reply:
Ah interesting, I'm not sure what the syntax in Scala 3 is supposed to be, but it seems that you would need something like `val using Quotes` to  make the implicit parameter public. The compiler error is likely showing up because the type of `classDynamicSymbol` depends on the implicit `Quotes`.

Reply:
I tried to find how to combine `val`s with `using Quotes` in the documentation and in the code samples on github but I didn't find any example of how to do this and the most typical construction still is `def someMethod(using Quotes)`. In addition, we can reduce the lines of code (and remove imports) in `def`s by rewriting in this way:
```
  def classDynamicSymbol(using quoted: Quotes) = 
    quoted.reflect.Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
```

Reply:
Oh, I guess at first time I didn't understand you completly with this `val using Quotes` but now I found the way how we can rewrite it cleaner. So here is how we can do:
```
class Helper[Q <: Quotes, T: Type](using val q: Q) {
   import q.reflect.*
   val classDynamicSymbol = Symbol.requiredClass("me.shadaj.scalapy.py.Dynamic")
}
```
But in the `native_impl` and `native_named_impl` in this case we will create the class object (don't know if it matters for us or no) during every `native_impl` and `native_named_impl` call. This is the creation of an object of class Helper in `native_imple`: 
```
  def native_impl[T: Type](using q: Quotes): Expr[T] = {
    import q.reflect.*
    val helper = new Helper[q.type, T]
```